### PR TITLE
fix(core): Fall back to the instance key when the key store is unavailable (no-changelog)

### PR DESCRIPTION
**Pull request by afitzek**
## Summary

Reading old (pre-rotation) data through the key-rotation feature routes every decryption of a legacy row through the key store. This change lets that read fall back to the instance key directly when the store cannot serve the legacy key — the key is not seeded, or the store is unreachable.

Old data is plain AES-256-CBC encrypted with the instance key, and the seeded legacy key is the instance key, so the fallback yields the same result the store path would. Reads of old data keep working while the store is degraded, so the read-support step does not add a new way to break reading old data.

This is part of the Encryption Key Rotation V2 rollout (Release A — Read-support). It is gated by `N8N_ENV_FEAT_ENCRYPTION_KEY_ROTATION` and produces no user-facing change on its own.

### How to test manually

Prerequisites: a running instance with a database you can edit.

1. Start n8n with `N8N_ENV_FEAT_ENCRYPTION_KEY_ROTATION` unset, then create a credential. This writes old-format data (AES-256-CBC under the instance key, no keyId prefix).
2. Set `N8N_ENV_FEAT_ENCRYPTION_KEY_ROTATION=true` and restart. The module loads and seeds the legacy key. Open the credential and confirm it still decrypts (this exercises the store path).
3. Delete the legacy key row: `DELETE FROM deployment_key WHERE type = 'data_encryption' AND algorithm = 'aes-256-cbc';`
4. Open the credential again. It still decrypts — the read falls back to the instance key. A `warn` log records the fallback.
5. Edge case — store outage: make the lookup fail (for example, stop the database mid-read). Reading old data still returns the plaintext through the same fallback.

Automated coverage: `pnpm --filter n8n test key-manager.service.test.ts` covers the store-serves path, the unseeded fallback, the lookup-failure fallback, and that the instance-key fallback is built once and reused.

## Related Linear tickets, Github issues, and Community forum posts

closes https://linear.app/n8n/issue/IAM-1289

## Review / Merge checklist

- [X] I have seen this code, I have run this code, and I take responsibility for this code.
- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [X] Tests included. 
- [ ] PR Labeled with `Backport to Beta`, `Backport to Stable`, or `Backport to v1` (if the PR is an urgent fix that needs to be backported)

<a href="https://cubic.dev/pr/n8n-io/n8n/pull/37439?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/cli/src/modules/encryption-key-manager/key-manager.service.ts [hunk 1]
```diff
@@ -42,15 +47,31 @@ export class KeyManagerService {
...
+	 * same result. This keeps reads of old data working while the store is degraded.
+	 */
 	async getLegacyKey(): Promise<KeyInfo> {
-		const key = await this.deploymentKeyRepository.findOne({
-			where: { type: 'data_encryption', algorithm: 'aes-256-cbc' },
-		});
-		if (!key) {
-			throw new NotFoundError('No legacy aes-256-cbc encryption key found');
+		try {
+			const key = await this.deploymentKeyRepository.findOne({
```
[the remark this answers is not in the document]
Comment (afitzek) [reply packages/cli/src/modules/encryption-key-manager/key-manager.service.ts:61]: This derivation is by design.

**Review by guillaumejacquart — APPROVED** [review]
makes sense, lgmt
